content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+
+image::/images/post-images/2017-01-17/freestyle-conditional-long-full.png[Longer Freestyle Job, role="center", width=800]
+
+
+[[longer-pipeline]]
+=== Pipeline
+
+Here's the equivalent Pipeline:
+
+.Pipeline
+[pipeline]
+----
+// Script //
+properties ([
+    parameters ([
+        string (
+            defaultValue: '*',
+            description: '',
+            name : 'BRANCH_PATTERN'),
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    ])
+])
+
+node {
+    stage ('Build') {
+        checkout([$class: 'GitSCM',
+            branches: [[name: "origin/${BRANCH_PATTERN}"]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [[$class: 'LocalBranch']],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: '9f926524-9fac-44df-99dd-53069380f286', url: 'https://github.com/bitwiseman/hermann']]])
+
+        GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+        if (GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD) {
+
+            parallel (
+                linux: {
+                    build job: 'full-build-linux', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                mac: {
+                    build job: 'full-build-mac', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                windows: {
+                    build job: 'full-build-windows', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                },
+                failFast: false)
+
+        } else {
+            echo 'Skipped full build.'
+        }
+    }
+}
+// Declarative //
+pipeline {
+    agent any
+    parameters {
+        string (
+            defaultValue: '*',
+            description: '',
+            name : 'BRANCH_PATTERN')
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    }
+
+    stages {
+        stage ('Prepare') {
+            steps {
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "origin/${BRANCH_PATTERN}"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'LocalBranch']],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[credentialsId: '9f926524-9fac-44df-99dd-53069380f286', url: 'https://github.com/bitwiseman/hermann']]])
+            }
+        }
+
+        stage ('Build') {
+            when {
+                expression {
+                    GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+                    return GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD
+                }
+            }
+            steps {
+                parallel (
```

Comment:
Right, but it's rather highlighting the "Look, no need for other jobs" advantage of Pipeline while at the same time adding lines like crazy, undermining the "it's simple" argument.

Maybe could just call build(…), then explain afterwards that this isn't even needed and it all can be one thing.